linode/terraform-provider-linode
# Include resource and data source name as a comment in user-agent

## 📝 Description

This is to include resource or data source name in the user agent of the API calls.

## ✔️ How to Test

### Automated Testing

Which runs very looooong and may have intermittent errors.
```bash
make test
```

### Manual Testing

- Enable logging:
```bash
export TF_LOG=DEBUG
export TF_LOG_PATH=$PWD/log.txt 
export TF_LOG_PROVIDER_LINODE_REQUESTS=DEBUG
```
- Deploy some resources, like:
```terraform
resource "linode_vpc" "test" {
    label = "test-vpc${count.index}"
    region = "us-mia"
    description = "My first VPC."
    count = 3
}

resource "linode_nodebalancer" "foobar" {
    label = "mynodebalancer${count.index}"
    region = "us-mia"
    client_conn_throttle = 20
    count = 3
}
```
- Inspect `log.txt` for user agents recorded, they should be something like `User-Agent="HashiCorp Terraform/1.8.1 (+https://www.terraform.io) Terraform-Plugin-Framework/1.8.0 terraform-provider-linode/dev (linode_vpc resource)"`
